src/server/list_family.cc

```diff
@@ -565,6 +563,62 @@ void ListFamily::LLen(CmdArgList args, ConnectionContext* cntx) {
   }
 }
 
+void ListFamily::LPos(CmdArgList args, ConnectionContext* cntx) {
+  string_view key = ArgS(args, 1);
+  string_view elem = ArgS(args, 2);
+
+  int rank = 1;
+  uint32_t count = 1;
+  uint32_t max_len = 0;
+
+  for (size_t i = 3; i < args.size(); i++) {
+    ToUpper(&args[i]);
+    const auto& arg_v = ArgS(args, i);
+    if (arg_v == "RANK") {
+      if (!absl::SimpleAtoi(ArgS(args, (i + 1)), &rank) || rank == 0) {
+        return (*cntx)->SendError(kInvalidIntErr);
+      }
+    }
+    if (arg_v == "COUNT") {
+      if (!absl::SimpleAtoi(ArgS(args, (i + 1)), &count)) {
+        return (*cntx)->SendError(kInvalidIntErr);
+      }
+    }
+    if (arg_v == "MAXLEN") {
+      if (!absl::SimpleAtoi(ArgS(args, (i + 1)), &max_len)) {
+        return (*cntx)->SendError(kInvalidIntErr);
+      }
+    }
+  }
+
+  auto cb = [&](Transaction* t, EngineShard* shard) {
+    return OpPos(t->GetOpArgs(shard), key, elem, rank, count, max_len);
+  };
+
+  Transaction* trans = cntx->transaction;
+  OpResult<vector<uint32_t>> result = trans->ScheduleSingleHopT(std::move(cb));
+
+  if (result.status() == OpStatus::WRONG_TYPE) {
+    return (*cntx)->SendError(result.status());
+  } else if (result.status() == OpStatus::INVALID_VALUE) {
+    return (*cntx)->SendError(result.status());
+  }
+
+  switch (result->size()) {
```

Comment:
I just went over the spec https://redis.io/commands/lpos/ and it seems that the output is determined by whether "count" is given and not by the cardinality of the result. 
In other words, it should always be an array if "count" is passed. even we used "count 1", for example, or the result does not have any matches.

Reply:
fixed, squashing commits..